students/johnpharmd/lesson07/html_render.py

```diff
@@ -0,0 +1,166 @@
+
+
+class Element():
+    """renders an html element"""
+
+    tag = ''
+    indent = '    '
+
+    def __init__(self, content=None, **kwargs):
+        self.content = [content]
+        self.kwargs = kwargs
+
+    def append(self, new_string):
+        return self.content.append(new_string)
+
+    def render(self, file_out, cur_ind=''):
+        if self.kwargs:
+            self.open_tag = '<' + self.tag
+            kwargs_string = ''
+            for k, v in self.kwargs.items():
+                kwargs_string += ' ' + k + '=' + '\'' + v + '\''
+            self.open_tag += kwargs_string + '>'
+            if self.tag == 'ul':
+                self.open_tag += '\n'
+        else:
+            if self.tag != 'a':
+                self.open_tag = '<' + self.tag + '>'
+            else:
+                self.open_tag = '<' + self.tag
+        self.close_tag = '</' + self.tag + '>'
+        if self.tag == 'html':
+            file_out.write(self.open_tag + '\n')
+        elif self.tag == 'ul':
+            file_out.write(cur_ind * 2 + self.open_tag)
+        elif self.tag == 'li':
+            file_out.write(cur_ind * 3 + self.open_tag + '\n')
+        elif self.tag == 'a':
+            file_out.write(cur_ind * 4 + self.open_tag)
+        elif self.tag != 'p':
+            file_out.write(cur_ind + self.open_tag + '\n')
```

Comment:
you should consider putting custom indentation logic into the respective child classes instead of here